keps/sig-scheduling/4832-async-preemption/README.md

```diff
@@ -0,0 +1,410 @@
+3. The scheduling cycle ends and the binding cycle starts; the binding cycle is run asynchronously, and the scheduler starts the next scheduling cycle.
+
+This flow allows us to decouple the API call to assign Pod to the Node from the scheduling cycle so that the API call doesn't block the scheduling throughput.
+
+But, we have the similar problem with the preemption; the preemption is run at PostFilter extension point which is the part of the scheduling cycle.
+The preemption has to make some API calls to update Pods' condition and delete Pods after all, which could block the scheduling throughput.
+
+scheduler-perf [actually shows](https://github.com/kubernetes/kubernetes/blob/342da505bdefbd849b808cca3cb76c24a993025f/test/integration/scheduler_perf/config/performance-config.yaml#L641) currently the preemption scenario takes too long time, compared to others.
+
+### Goals
+
+- Improve scheduling throughput when pods require issuing preemptions by making API calls asynchronous
+
+### Non-Goals
+
+- Making the same enhancement for DRA is not a goal of this KEP because it's an under-construction feature yet.
+  - If DRA maintainers want, technically they can along with this KEP. But, at least in this KEP, we don't discuss how.
+
+## Proposal
+
+The preemption plugin makes API calls for the preemption asynchronously after `PostFilter` extension point 
+so that the scheduler can continue to other Pods' scheduling while making API calls for preemption.
+After the preemption goroutine is done, the scheduling for the Pod that triggered the preemption will be retried.
+
+### Risks and Mitigations
+
+#### When kube-apiserver is unstable
+
+When kube-apiserver is unstable and API calls at the preemption goroutine fails frequently,
+the scheduler could make a non-optimal scheduling decision 
+because the scheduler nominates pods at `PostFilter` though, those Pods won't be scheduled on nodes because the preemption API calls fail.
+
+Let's say many mid-priority Pods are making the preemption API calls.
+With the scheduler after this proposal, during the preemption goroutine for them are runnning, 
+the scheduler assumes they'll be scheduled at the Nodes eventually
+that the preemptions are targeting via `.Status.NominatedNodeName`.
+So, other mid-priority or lower priority Pods' scheduling take those preemptor Pods into consideration, 
+which is correct if the preemption goroutine finishes successful actually, while which results in non-best scheduling results otherwise.
+(Higher priority Pods won't be affected; Pods can take place of reserved for lower priority Pods via `.Status.NominatedNodeName`)
+
+But, in the first place though, when kube-apiserver is unstable, the scheduler doesn't behave well 
+because it works with a lot of communication with kube-apiserver.
+Even if the scheduler makes the best scheduling result, the binding API might fail after all.
+
+So, we don't have to pay a special attention to this issue.
```

Comment:
I don't fully agree with this. We have to pay attention to it, but it's orthogonal to this KEP.

Reply:
How would you change the sentence? (From my non-native-speaker eyes, `a "special" attention` delivers the nuance.)